loleaflet/src/layer/tile/CanvasSectionContainer.ts

```diff
@@ -285,10 +311,205 @@ class CanvasSectionContainer {
 		}
 	}
 
+	private updateBoundSectionList(section: CanvasSectionObject, sectionList: Array<CanvasSectionObject> = null): Array<CanvasSectionObject>{
```

Comment:
Can we have a non recursive version of this please, which will be much easier to understand and write too. A tweaked double for-loop can achieve this I think.

Reply:
Hmm, i think we can't achieve this functionality with double for loop. Think complicated cases, there may be many sections bound to each other.

Reply:
Actually, we can do even better, in one linear pass we use a map to get the "section->parent_of_boundsection_if_any" relationship and use this map to create lists of bound sections if it is really needed.

Reply:
I see you mean something like std::map right?
I put these lists because it would need to be re-calculated for every event, so lists are created performance in mind.
boundsList is an ordered list.
Bound sections may be complicated, i really think that we need recursive function to achieve this goal.
parent_of_boundsection does not have to be on top of that section, and sections do not need to be bound to one single section. There may be a view like a family tree. Current implementation allows this.
I think this function is fine.

Reply:
I see, so "binding" of sections are bidirectional and don't have a parent child relationship. So to generalize this situation am I correct with this statement ? If we have a N sections, the job is to find disjoint sets of sections such that each section in a set is bound to every other section in that set. After finding those sets we need to order them by zindex and drawing order. If this is true, then there is a super fast way to achieve this using a famous data structure called "union-find" (Refer https://en.wikipedia.org/wiki/Disjoint-set_data_structure). This data structure is dynamic, so we can add/remove sections or change bindings of any section. For our purpose, we have to do the sorting of members of the sets ourselves.

As this requires more time implementing lets get this in now and hopefully come back to replace that costly recursive function called in a loop some day :)

Some side comments:
The lists/arrays in JS may have efficient implementations, but an algorithm >= O(N^2) can be made to crawl very easily (much easily for an exponential case algo) even if it uses super optimized data structure implementations and/or run on super fast hardware. Recursive functions may look neat and compact but it can get incredibly hard to debug them when we hit a problem with it. I used to write recursive functions until I had to debug them :)